rafthttp/stream.go

```diff
@@ -323,13 +326,15 @@ func (cr *streamReader) run() {
 			}
 		}
 		select {
-		// Wait 100ms to create a new stream, so it doesn't bring too much
-		// overhead when retry.
-		case <-time.After(100 * time.Millisecond):
 		case <-cr.stopc:
 			plog.Infof("stopped streaming with peer %s (%s reader)", cr.peerID, t)
 			close(cr.done)
 			return
+		default:
+			// wait for a while before new dial attempt
+			if err := cr.rl.Wait(context.TODO()); err != nil {
```

Comment:
not a fan of this forced timeout, would it be possible to replace `stopc` with
```go
type streamReader struct {
...
    runCtx context.Context
    runCancel context.CancelFunc
...
}
```

and replace the `select` with:
```go
err := cr.rl.Wait(cr.runCtx)
if cr.runCtx.Err() != nil {
    plog.Infof("...")
    close(cr.done)
    return
}
if err != nil {
    plog.Errorf(...)
}
```
and replace any `<-stopc` with `<-runCtx.Done()`
?